### DISCUSSION TITLE
Custom Logger in Netty

### DISCUSSION THREAD

**Discussion by isaacrivriv** [netty/netty#12335] [General]
Hey there! First time starting a discussion here. Just starting out diving into Netty and I was curious about implementing my own Logger reusing a custom logger. Was seeing the [Log4J2Logger](https://netty.io/4.1/api/io/netty/util/internal/logging/Log4J2LoggerFactory.html) and was thinking of doing something similar to that implementation but looking into the interface for [InternalLogger](https://netty.io/4.1/api/io/netty/util/internal/logging/InternalLogger.html) I see a message saying **`Internal-use-only logger used by Netty. DO NOT access this class outside of Netty`**

Is it possible to implement and add your own logger to Netty? Are there any caveats or issues if so?

Thanks in advance!

**Comment by chrisvest** [comment]
It's possible. The main caveat is, since this is internal, we will be at liberty to change the interface in patch releases.
Another caveat is noted in the `InternatLoggerFactory` javadoc: https://netty.io/4.1/api/io/netty/util/internal/logging/InternalLoggerFactory.html

**Reply by isaacrivriv** [reply]
Thanks for the quick response! Wasn't aware of that caveat, I'll look more into this.
